tests/Cms/Media/MediaTest.php

```diff
@@ -106,13 +126,144 @@ public function testPublish()
         $this->assertFalse(is_dir($versionB1));
     }
 
+    /**
+     * @covers ::root
+     */
+    public function testRoot()
+    {
+        // copy test image to content
+        Dir::make($this->tmp . '/content');
+        F::copy($this->fixtures . '/../files/test.jpg', $this->tmp . '/content/test.jpg');
+
+        $kirby = $this->app;
+
+        // string model
+        $hash = '12345678';
+        $this->assertStringEndsWith('tests/Cms/Media/tmp/media/assets/assets/' . $hash, Media::root($kirby, 'assets', $hash));
+
+        // file object
+        $file  = $kirby->file('test.jpg');
+        $hash  = $file->mediaHash();
+        $this->assertStringEndsWith('tests/Cms/Media/tmp/media/site/' . $hash, Media::root($kirby, $file, $hash));
+
+        // site object
+        $site  = $kirby->site();
+        $hash  = $file->mediaHash();
+        $this->assertStringEndsWith('tests/Cms/Media/tmp/media/site/' . $hash, Media::root($kirby, $site, $hash));
+    }
+
+    /**
+     * @covers ::thumb
+     */
+    public function testThumb()
+    {
+        Dir::make($this->tmp . '/content');
+
+        // copy test image to content
+        F::copy($this->fixtures . '/../files/test.jpg', $this->tmp . '/content/test.jpg');
+
+        // get file object
+        $file  = $this->app->file('test.jpg');
+        Dir::make(dirname($file->mediaRoot()));
+        $this->assertInstanceOf('\Kirby\Cms\File', $file);
+
+        // invalid with empty job file
+        F::write(dirname($file->mediaRoot()) . '/.jobs/' . $file->filename() . '.json', '{}');
+        $thumb = Media::thumb($file, $file->mediaHash(), $file->filename());
+        $this->assertFalse($thumb);
+
+        // create job file
+        $jobString = '{"width":64,"height":64,"quality":null,"crop":"center","filename":"test.jpg"}';
+        F::write(dirname($file->mediaRoot()) . '/.jobs/' . $file->filename() . '.json', $jobString);
+
+        // invalid with file not found
+        $thumb = Media::thumb($file, $file->mediaHash(), $file->filename());
+        $this->assertInstanceOf('Kirby\Cms\Response', $thumb);
+        $this->assertSame('', $thumb->body());
```

Comment:
I know this comes from the existing test, but I wonder: Why does this example lead to an empty response?